compact_test.go

```diff
@@ -1058,3 +1061,33 @@ func TestDeleteCompactionBlockAfterFailedReload(t *testing.T) {
 		})
 	}
 }
+
+func TestOpenBlockWithHook(t *testing.T) {
+	//init action
+	original := filepath.Join(string(filepath.Separator), "tmp", fmt.Sprintf("dev-%d", time.Now().Unix()))
+	mountpoint := filepath.Join(string(filepath.Separator), "tmp", fmt.Sprintf("mountpoint-%d", time.Now().Unix()))
```

Comment:
something like this would be a bit better
```
original, err := ioutil.TempDir("", "test")
testutil.Ok(t, err)
mountpoint, err := ioutil.TempDir("", "test")
testutil.Ok(t, err)
```

than can remove the creation logix in the NewServer.

Reply:
Accept and updated